capsules/src/process_console.rs

```diff
@@ -1024,6 +1036,51 @@ impl<'a, C: ProcessManagementCapability> ProcessConsole<'a, C> {
             Ok(())
         }
     }
+
+    /// If there is anything in the queue, copy it to the TX buffer and send
+    /// it to the UART.
+    ///
+    /// Returns Ok(usize) with the number of bytes sent from the queue. If Ok(0)
+    /// is returned, nothing was sent and the UART is free.
+    fn handle_queue(&self) -> Result<usize, ErrorCode> {
+        if self.tx_in_progress.get() {
+            // This shouldn't happen because we should only try to handle the
+            // queue when nothing else is happening, but still have the check
+            // for safety.
+            return Err(ErrorCode::BUSY);
+        }
+
+        self.queue_buffer.map_or(Err(ErrorCode::FAIL), |qbuf| {
+            let qlen = self.queue_size.get();
+
+            if qlen > 0 {
+                self.tx_buffer.take().map_or(Err(ErrorCode::FAIL), |txbuf| {
+                    let txlen = cmp::min(qlen, txbuf.len());
+
+                    // Copy elements of the queue into the TX buffer.
+                    (&mut txbuf[..txlen]).copy_from_slice(&qbuf[..txlen]);
+
+                    // TODO: If the queue needs to print over multiple TX
+                    // buffers, we need to shift the remaining contents of the
+                    // queue back to index 0.
+                    // if qlen > txlen {
+                    //     (&mut qbuf[txlen..qlen]).copy_from_slice(&qbuf[txlen..qlen]);
+                    // }
```

Comment:
Do we still need to address this?

This looks like something that should use `core::slice::copy_within()`, if the issue is just that this would not compile.

Reply:
Yes this would be an improvement, in case we cannot send the entire queued buffer.